ui/components/multichain/menu-items/view-explorer-menu-item.tsx

```diff
@@ -14,34 +18,51 @@ import {
   MetaMetricsEventName,
 } from '../../../../shared/constants/metametrics';
 import { IconName, Text } from '../../component-library';
-import {
-  getBlockExplorerLinkText,
-  getCurrentChainId,
-  getRpcPrefsForCurrentProvider,
-} from '../../../selectors';
+import { getBlockExplorerLinkText } from '../../../selectors';
 import { getURLHostName } from '../../../helpers/utils/util';
 import { NETWORKS_ROUTE } from '../../../helpers/constants/routes';
-import { toChecksumHexAddress } from '../../../../shared/modules/hexstring-utils';
+import { getMultichainNetwork } from '../../../selectors/multichain';
+import { useMultichainSelector } from '../../../hooks/useMultichainSelector';
+
+export type ViewExplorerMenuItemProps = {
+  /**
+   * Represents the "location" property of the metrics event
+   */
+  metricsLocation: string;
+  /**
+   * Closes the menu
+   */
+  closeMenu?: () => void;
+  /**
+   * Custom properties for the menu item text
+   */
+  textProps?: object;
+  /**
+   * Account to show account details for
+   */
+  account: InternalAccount;
+};
 
 export const ViewExplorerMenuItem = ({
   metricsLocation,
   closeMenu,
   textProps,
-  address,
-}) => {
+  account,
+}: ViewExplorerMenuItemProps) => {
   const t = useI18nContext();
   const trackEvent = useContext(MetaMetricsContext);
   const history = useHistory();
 
-  const chainId = useSelector(getCurrentChainId);
-  const rpcPrefs = useSelector(getRpcPrefsForCurrentProvider);
-  const addressLink = getAccountLink(
-    toChecksumHexAddress(address),
-    chainId,
-    rpcPrefs,
+  const multichainNetwork = useMultichainSelector(
+    getMultichainNetwork,
+    account,
   );
-
-  const { blockExplorerUrl } = rpcPrefs;
+  const addressLink = getMultichainAccountUrl(
+    account.address,
+    multichainNetwork,
+  );
+  const chainId = parseCaipChainId(multichainNetwork.chainId).reference;
```

Comment:
Ok thanks for the clarification. I would add a `TODO` here to convert this back to a CAIP-2 one day:

```suggestion
  // TODO: Re-use CAIP-2 for metrics once event schemas support it
  const chainId = parseCaipChainId(multichainNetwork.chainId).reference;
```